core/domain/state_domain.py

```diff
@@ -2325,6 +2332,25 @@ def validate(self, exp_param_specs_dict, allow_null_interaction):
                     .get_content_ids()
                 )
 
+        if len(set(content_id_list)) != len(content_id_list):
+            raise utils.ValidationError(
+                'Expected all content_ids to be unique, '
+                'recieved %s' % content_id_list)
+
+        for content_id in content_id_list:
+            content_id_suffix = content_id.split('_')[-1]
+
+            # Possible values of content_id_suffix are a digit, or from
+            # a 'outcome' (from 'default_outcome'). If the content_id_suffix
+            # is not a digit, we disregard it here.
+            if (
+                    content_id_suffix.isdigit() and
+                    int(content_id_suffix) > self.next_content_id_index
+            ):
+                raise utils.ValidationError(
+                    'Expected all content id indexes to be less than next '
```

Comment:
than next content id index --> than the "next content id index"

Reply:
Done